docs/okeps/okep-5377-extend-udn-to-support-multiple-cluster-subnets-in-layer3-topology.md

```diff
@@ -0,0 +1,124 @@
+# OKEP-5377: Extend UDN to Support Multiple Cluster Subnets in Layer3 Topology 
+
+* Issue: [#5377](https://github.com/ovn-org/ovn-kubernetes/issues/5377)
+
+## Problem Statement
+
+UDN currently supports only one cluster subnet per IP family in Layer3 topology. This limits IP
+address planning and scalability for large clusters.
+
+## Goals
+
+* Support multiple cluster subnets per IP family in UDN's Layer3 topology.
+* Align UDN’s subnet capabilities with existing support in NAD.
+* Allow dynamic subnet expansion by appending new subnets to an existing UDN.
+* Support only subnets with the same host subnet size(`hostSubnet`).
+
+## Non-Goals
+
+* Supporting multiple subnets in Layer2 or Localnet topologies.
+* Supporting deletion or shrinking of existing subnets.
+
+## Future-Goals
+
+* Support subnets with different host subnet sizes(`hostSubnet`)
+
+## Introduction
+
+When using UDN to define a network in Layer3 topology, each node gets a subnet allocated from the
+UDN's `subnets` list per IP family. Currently, UDN limits `subnets` to a single subnet per IP
+family, enforced by CRD validation and full immutability of the UDN spec. This limitation blocks
+practical use cases where operators need to expand IP space incrementally — a feature already
+available through NAD.
+
+This proposal aims to bring UDN's flexibility in line with NAD by supporting multiple cluster
+subnets and enabling controlled mutability.
+
+
+## User-Stories/Use-Cases
+
+### Story 1: Expanding subnet pool for growing clusters
+
+**As a** cluster operator managing a UDN Layer3 network,  
+**I want** to add new subnets when existing ones near exhaustion,  
+**so that** I can scale the cluster without recreating the network.
+
+
+## Proposed Solution
+
+### API Details
+
+1. Update `UDN`/`CUDN` CRD to allow `layer3.subnets` to accept multiple subnets per IP family.
+2. Allow updates to the `layer3.subnets` field, but only to **append** new entries.
+   - Reject updates that attempt to remove or modify existing subnet entries.
+   - Reject new entries that overlap with existing subnets or use a different `hostSubnet`.
+
+
+### Implementation Details
+
```

Comment:
details of changes to the OVN-Kubernetes topology when you add the second and third subnets is not covered here in this OKEP.

Reply:
more details added, please take a look, please kindly let me know if anything is missing, thanks.